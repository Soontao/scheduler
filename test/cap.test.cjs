const cds = require("@sap/cds");
const path = require("path");

describe("CAP Base Test Suite", () => {

  const { axios } = cds.test(path.join(__dirname, ".."));
  axios.defaults.validateStatus = () => true;

  it('should support retrieve metadata', async () => {
    const response = await axios.get("/api/v1/job/$metadata");
    expect(response.data).toMatch(/Jobs/);
    expect(response.data).toMatch(/Tasks/);
  });

  it('should support create job', async () => {
    const response = await axios.post("/api/v1/job/Jobs", {
      name: "test_job_1",
      active: false,
      cron: "*/2 * * * *",
      tasks: [
        {
          name: "test_task_1"
        }
      ]
    });

    expect(response.status).toBe(201);

  });

  it('should support reject wrong cron pattern', async () => {
    const response = await axios.post("/api/v1/job/Jobs", {
      name: "test_job_2",
      active: false,
      cron: "*/2 * 40 * *",
      tasks: [
        {
          name: "test_task_2"
        }
      ]
    });

    expect(response.status).toBe(400);
    expect(response.data?.error).toMatchSnapshot();

  });

});