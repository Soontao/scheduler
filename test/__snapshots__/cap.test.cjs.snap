// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CAP Base Test Suite should support reject wrong cron pattern 1`] = `
{
  "@Common.numericSeverity": 4,
  "code": "400",
  "message": "Constraint error, got value 40 expected range 1-31",
  "target": "cron",
}
`;
